doc/source/ray-dashboard.rst

```diff
@@ -0,0 +1,157 @@
+If this number keeps increasing as you run your Ray cluster, that could mean that your objectIDs are not properly garbage collected. This can lead to OOM errors or eviction of objectIDs that your program still wants to use. 
+
+
+**Number of local objects are way less than number of object IDs in scope**
+This means many of objectIDs are referring to objects residing in plasma stores. It can happen either when objects are big or when you use ``ray.put``. 
+If you have too many big objects and if your program uses plasma store heaviily, the program could be slower becasue using plasma stores is slower than local memory.
+
+Profiling (Experimental)
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+.. note::
+
+  The profiling button currently only works when you use **passwordless** ``sudo``. 
+  It is still experimental. Please report any issues you run into.
+
+.. image:: https://raw.githubusercontent.com/ray-project/images/master/docs/dashboard/dashboard-profiling-buttons.png
+    :align: center
+
+Clicking one of the profiling buttons on the dashboard launches py-spy, which will profile your actor process for the given duration. Once the profiling has been done, you can click the "profiling result" button to visualize the profiling information as a flamegraph.
+This visualization can help reveal computational bottlenecks.
+
+More information on how to interpret the flamegraph is available at https://github.com/jlfwong/speedscope#usage.
+
+.. image:: https://raw.githubusercontent.com/ray-project/images/master/docs/dashboard/dashboard-profiling.png
+    :align: center
+
+Components
+-----------
+
+Machine View
+~~~~~~~~~~~~
+The machine view provides node and process level information. You can see resource usage statistics for each node and each worker process.
+
+Logical View (Experimental)
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+The logical view renders actor information in a tree format. To illustrate this, in the code block below, the ``Parent`` actor creates two ``Child`` actors and each ``Child`` actor creates one ``GrandChild`` actor.
+This relationship will be visible at a dashboard logical view page.
+
+.. code-block:: python
+
+  import ray
+  ray.init()
+
+  @ray.remote
+  class Grandchild:
+      def __init__(self):
+          pass
+
+  @ray.remote
+  class Child:
+      def __init__(self):
+          self.grandchild_handle = Grandchild.remote()
+  
+  @ray.remote
+  class Parent:
+      def __init__(self):
+          self.children_handles = [Child.remote() for _ in range(2)]
+
+  parent_handle = Parent.remote()
+
+You can see that the dashboard shows the parent/child relationship as expected. 
+
+.. image:: https://raw.githubusercontent.com/ray-project/images/master/docs/dashboard/dashboard-tree-actors.png
```

Comment:
Is it ok to actually move these images into the Ray repo? We currently do that with all of our other images.

Reply:
I suggested putting them in the other repo because I was concerned about their size. I imagine we'll want to add a bunch more images as time goes on, and my concern is that cloning the Ray repository will become too slow.

Reply:
I imagine this is an issue that a lot of projects run into, so there is probably some standard solution that I'm not aware of.

Reply:
Seems like the standard solution is to not have that many images :) (just checked OpenAI and pytorch repos, everyone bundles images together, but they don't have that many).

Reply:
Anyways, I'm not opposed to the idea, but I would suggest a 1 page proposal/memo for this practice to be shared with the Ray team (or else no one knows about it).

Reply:
Will create a proposal and write it to slack open-source channel.